src/Build/Logging/BinaryLogger/BinaryLogger.cs

```diff
@@ -175,20 +175,11 @@ public void Shutdown()
 
             if (projectImportsCollector != null)
```

Comment:
Nit:
```suggestion
            if (projectImportsCollector != null && CollectProjectImports == ProjectImportsCollectionMode.Embed)
```

Reply:
I think that would be wrong - we want to go into the if so that we call Close on 178 even for ZipFile not just for Embed

Reply:
Oh you're completely right. Whoops!